packages/world/src/runs.ts

```diff
@@ -12,22 +17,56 @@ export const WorkflowRunStatusSchema = z.enum([
   'cancelled',
 ]);
 
-export const WorkflowRunSchema = z.object({
+/**
+ * Base schema for the Workflow runs. Prefer using WorkflowRunSchema
+ * which implements a discriminatedUnion for various states
+ */
+export const WorkflowRunBaseSchema = z.object({
   runId: z.string(),
-  deploymentId: z.string(),
   status: WorkflowRunStatusSchema,
+  deploymentId: z.string(),
   workflowName: z.string(),
   executionContext: z.record(z.string(), z.any()).optional(),
   input: z.array(z.any()),
   output: z.any().optional(),
-  error: z.string().optional(),
-  errorCode: z.string().optional(),
+  error: StructuredErrorSchema.optional(),
   startedAt: z.coerce.date().optional(),
   completedAt: z.coerce.date().optional(),
   createdAt: z.coerce.date(),
   updatedAt: z.coerce.date(),
 });
 
+// Discriminated union based on status
+export const WorkflowRunSchema = z.discriminatedUnion('status', [
+  // Non-final states: output and error are optional
```

Comment:
code is right, comment isn't
```suggestion
  // Non-final states: output and error are undefined
```